### PR TITLE
fix: modernize class inheritance

### PR DESCRIPTION
**Pull request by jimmywarting**
Wanted to modernize it a bit

### CODE REVIEW

**Comment by wraithgar** [comment]
This is much more readable!  Unfortunately tests don't pass and the tests here are quite fiddly.  If you can get the entire refactor working without changing tests it makes it easier to land larger refactors like this.  If you can't, we may have to break this up into smaller changes that change both the code and the tests.  It's easier to reason about changes to both with smaller iterative changes.

**Comment by jimmywarting** [comment]
yea... trying to figure out what's wrong.
the problem with smaller iterative changes where that some classes extends others and dose something like `XYZ.call(this, ...args)` and that dose not quite work so well when you received a `XYZ` can not be constructed without using `new` keyword.

**Comment by jimmywarting** [comment]
I think it where just that the test threshold did get lower cuz it was using less code. so the uncovered lines covered more %
so i added a simple debug test to one of the test cases

**Comment by wraithgar** [comment]
I have no idea why `Pull Request / Lint Commits` is failing but that's our problem to fix, not yours.  The PR title itself looks fine to me?

File: lib/tracker-group.js [hunk 1]
```diff
@@ -1,116 +1,112 @@
...
+      tracker.removeListener('change', this.bubbleChange)
+    }
+    this.emit('change', this.name, 1, this)
+  }
 
-TrackerGroup.prototype.newStream = function (name, todo, weight) {
-  return this.addUnit(new TrackerStream(name, todo), weight)
-}
+  debug (depth = 0) {
+    const indent = ' '.repeat(depth)
```
[the remark this answers is not in the document]
Comment (lukekarrys) [reply lib/tracker-group.js:90]: Good catch. I think this is fine tbh and not worth mitigating against. Users would have to create a lot of nested trackers for it to become any kind of issue.

**Review by wraithgar — APPROVED** [review]

**Comment by wraithgar** [comment]
Cheers @jimmywarting

**Pull request merged**